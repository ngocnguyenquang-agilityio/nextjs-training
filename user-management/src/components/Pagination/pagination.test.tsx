import { RenderResult, render, screen } from '@testing-library/react';

// Components
import { Pagination } from '.';

const props = {
  totalPages: 4,
  standingPage: '1'
};

describe('Pagination Component', () => {
  let comp: RenderResult;

  beforeEach(() => {
    comp = render(<Pagination {...props} />);
  });

  test('Should render successfully', () => {
    expect(comp).toBeTruthy();
  });

  test('Should match snapshot', () => {
    expect(comp).toMatchSnapshot();
  });

  test('Should render correct number of pagination', async () => {
    const items = await screen.getByTestId('pagination');
    expect(items.children).toHaveLength(4);
  });

  test('Should get correct standing page', async () => {
    const items = await screen.getByTestId('pagination');
    expect(items.getElementsByClassName('bg-blue-600').length).toBe(1);
  });

  //TODO: Testing onClick to change pagination
});
