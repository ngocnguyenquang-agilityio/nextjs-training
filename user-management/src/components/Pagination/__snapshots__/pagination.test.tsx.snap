// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="pt-4 flex justify-end"
      >
        <ul
          class="inline-flex -space-x-px text-md gap-4"
          data-testid="pagination"
        >
          <li>
            <button
              class="flex items-center text-sm font-md bg-blue-500 text-white border border-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
              value="1"
            >
              1
            </button>
          </li>
          <li>
            <button
              class="flex items-center text-sm font-md bg-transparent text-blue-500 border border-blue-500 transition-colors hover:bg-blue-400 hover:text-white focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
              value="2"
            >
              2
            </button>
          </li>
          <li>
            <button
              class="flex items-center text-sm font-md bg-transparent text-blue-500 border border-blue-500 transition-colors hover:bg-blue-400 hover:text-white focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
              value="3"
            >
              3
            </button>
          </li>
          <li>
            <button
              class="flex items-center text-sm font-md bg-transparent text-blue-500 border border-blue-500 transition-colors hover:bg-blue-400 hover:text-white focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
              value="4"
            >
              4
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="pt-4 flex justify-end"
    >
      <ul
        class="inline-flex -space-x-px text-md gap-4"
        data-testid="pagination"
      >
        <li>
          <button
            class="flex items-center text-sm font-md bg-blue-500 text-white border border-blue-500 transition-colors hover:bg-blue-400 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
            value="1"
          >
            1
          </button>
        </li>
        <li>
          <button
            class="flex items-center text-sm font-md bg-transparent text-blue-500 border border-blue-500 transition-colors hover:bg-blue-400 hover:text-white focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
            value="2"
          >
            2
          </button>
        </li>
        <li>
          <button
            class="flex items-center text-sm font-md bg-transparent text-blue-500 border border-blue-500 transition-colors hover:bg-blue-400 hover:text-white focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
            value="3"
          >
            3
          </button>
        </li>
        <li>
          <button
            class="flex items-center text-sm font-md bg-transparent text-blue-500 border border-blue-500 transition-colors hover:bg-blue-400 hover:text-white focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-blue-500 active:bg-blue-600 px-4 py-2 rounded-md"
            value="4"
          >
            4
          </button>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
